### PR TITLE
makes gister auto-commit dirty gist repos if the GISTER_AUTO_COMMIT e…

### PR DESCRIPTION
**Pull request by IRobL**
…nv var is set to anything

### CODE REVIEW

File: gister.sh [hunk 1]
```diff
@@ -308,7 +308,13 @@ sync_gist() {
           legit sync > /dev/null
         else
           if (is_dirty); then
-            echo "DIRTY $gist_id"
+            if [ -z $GISTER_AUTO_COMMIT ]; then
+              echo "DIRTY $gist_id"
+            else
+              echo "DIRTY $gist_id, auto committing"
+              git add . && git commit -m "automated update" && \
```
Comment (weakish) [inline gister.sh]: `&& \` is unnecessary. Just use `&&`.

Comment (TheNotary) [reply gister.sh]: Good catch, thanks!

Comment (TheNotary) [reply gister.sh]: I was meaning to escape the end of line character which I thought was necisary, but it's probably best to have them as separate lines anyway.

File: gister.sh [hunk 2]
```diff
@@ -312,8 +312,8 @@ sync_gist() {
               echo "DIRTY $gist_id"
             else
               echo "DIRTY $gist_id, auto committing"
-              git add . && git commit -m "automated update" && \
-                git pull > /dev/null && git push > /dev/null
+              git add . && git commit -m "automated update"
```
Comment (weakish) [inline gister.sh]: I think here `&&` is necessary. If for some reason `git commit` fails, we should not pull/push.

**Pull request merged**